File: Programing_Basics_with_JS/conditionalStatementsMoreExercises/02.SleepyTomCat.js
function sleepyTomCat(input) {
    let holidays = Number(input[0]);

    let tomNormGameMinutes = 30000;
    let allDays = 365;

    let workingDays = allDays - holidays;
    let playingMinutes = workingDays * 63 + holidays * 127;

    let timeDiff = Math.abs(tomNormGameMinutes - playingMinutes);
    let hours = Math.floor(timeDiff / 60);
    let minutes = timeDiff % 60;


    if (tomNormGameMinutes > playingMinutes) {
        console.log('Tom sleeps well');
        console.log(`${hours} hours and ${minutes} minutes less for play`);
    } else {
        console.log('Tom will run away');
        console.log(`${hours} hours and ${minutes} minutes more for play`);
    }
}

sleepyTomCat(['20'])
sleepyTomCat(['113'])

File: Programming_Basics_with_Python/conditional_statements_more_exercise/02._sleepy_tom_cat.py
def tom_sleep(free_days, year_playing_minutes, work_day_playing_minutes, free_day_playing_minutes, days_in_year):
    work_days = days_in_year - free_days
    total_playing_in_minutes = (work_day_playing_minutes * work_days) + (free_day_playing_minutes * free_days)
    difference_between_hours = year_playing_minutes - total_playing_in_minutes

    if total_playing_in_minutes > year_playing_minutes:
        return f'Tom will run away \n{abs(difference_between_hours) // 60} hours and {abs(difference_between_hours) % 60} minutes more for play'
    else:
        return f'Tom sleeps well \n{difference_between_hours // 60} hours and {difference_between_hours % 60} minutes less for play'


free_days = int(input())

year_playing_minutes = 30000
work_day_playing_minutes = 63
free_day_playing_minutes = 127
days_in_year = 365

print(tom_sleep(free_days, year_playing_minutes, work_day_playing_minutes, free_day_playing_minutes, days_in_year))
